test/bigtest/tests/find-package-test.js

```diff
@@ -231,4 +235,199 @@ describe('ui-plugin-find-package-title', function () {
+                      selectedCount: 0,
+                      titleCount: 1,
+                      visibilityData: {
+                        isHidden: false,
+                        reason: ''
+                      }
+                    },
+                    {
+                      id: '440-3392',
+                      type: 'packages',
+                      contentType: 'Abstract and Index',
+                      customCoverage: {
+                        beginCoverage: '',
+                        endCoverage: ''
+                      },
+                      isCustom: false,
+                      isSelected: true,
+                      name: 'Selected also',
+                      packageId: 3391,
+                      packageType: 'Complete',
+                      providerId: 440,
+                      providerName: 'Getty Conservation Institute',
+                      selectedCount: 0,
+                      titleCount: 1,
+                      visibilityData: {
+                        isHidden: false,
+                        reason: ''
+                      }
+                    },
+                  ])).to.be.true;
+                });
+              });
+            });
+          });
+        });
+      });
+
+      describe('when searching for all packages', () => {
+        beforeEach(async () => {
+          await plugin.modal.searchField.fill('selected');
+        });
+
+        describe('and submit button was clicked', () => {
+          beforeEach(async () => {
+            await plugin.modal.searchButton.click();
+          });
+
+          it('should show correct number of packages', () => {
+            expect(plugin.modal.resultsList.rows().length).to.equal(3);
+          });
+
+          describe('and select all checkbox was checked', () => {
+            beforeEach(async () => {
+              await plugin.modal.clickSelectAllCheckbox();
+            });
+
+            describe('and clicking Save', () => {
+              beforeEach(async () => {
+                await plugin.modal.saveSelection();
+              });
+
+              it('should call the provided callback with correct data', () => {
+                const packages = onRecordChosenHandler.getCall(0).args[0];
+                expect(packages.length).to.equal(3);
+              });
+            });
+
+            describe('when entering another search query', () => {
+              beforeEach(async () => {
+                await plugin.modal.searchField.fill('ebsco');
+              });
+
+              describe('and submit button was clicked', () => {
+                beforeEach(async () => {
+                  await plugin.modal.searchButton.click();
+                });
+
+                describe('and select all checkbox was checked', () => {
+                  beforeEach(async () => {
+                    await plugin.modal.clickSelectAllCheckbox();
+                  });
```

Comment:
Could these all `beforeEach` be combined? Because as I remember the one of the failing tests is a big nesting of them.